File: snippets/multi-item-params.pl
#!/usr/bin/env perl
# -*- Mode: Perl; tab-width: 2; indent-tabs-mode: nil -*- vim:sta:et:sw=2:ts=2:syntax=perl
#

my @servers = ();
use Getopt::Long;

GetOptions(
  's|server|hosts=s' => \@servers
) or pod2usage(2);

@servers = split(/[,; ]+/, join(',', @servers));

File: snippets/multi-item-params.py
#!/usr/bin/env python
# -*- Mode: Python; tab-width: 2; indent-tabs-mode: nil -*-

import sys
from optparse import OptionParser


def main():
    usage = "usage: %prog [options]"
    parser = OptionParser(usage)
    parser.add_option(
        "-s", "--server", action="append", dest="servers", help="specify servers"
    )

    (opts, _args) = parser.parse_args()
    options = vars(opts)
    print(options)

    if options["servers"]:
        sep = ","
        joined = sep.join(options["servers"])
        servers = joined.split(sep)
    else:
        servers = None
    print(servers)


if __name__ == "__main__":
    main()
